app/_kong_plugins/solace-consume/index.md

```diff
@@ -0,0 +1,149 @@
+      All sessions on a plugin share the same event loop. The plugin executes non-blocking event handling to avoid delaying {{site.base_gateway}} requests.
+
+      {:.info}
+      > **Note:** Make sure that your ulimits are high enough to handle the number of concurrent connections, and ensure that clients are using keepalive connections to avoid reaching the maximum number of file descriptors.
+
+---
+
+This plugin allows {{site.base_gateway}} to consume messages from a [Solace PubSub+ Event Broker](https://solace.com/products/event-broker/) and makes them available through HTTP endpoints.
+For more information, see [Understanding Solace topics](https://docs.solace.com/Get-Started/what-are-topics.htm).
+
+The Solace Consume plugin includes the following features:
+* Supports Basic and OAuth2 Authentication for secure communication with Solace
+* Allows dynamic configuration of Solace session properties
+* Supports custom content payloads or direct payload forwarding
+* Provides configurable acknowledgment wait time to handle guaranteed message delivery
+* SSL/TLS support for secure connections to Solace brokers
+
+Kong also provides Solace plugins for logging and publishing messages:
+* [Solace Log](/plugins/solace-log/)
+* [Solace Upstream](/plugins/solace-upstream/)
+
+## Use cases and examples
+
+{% table %}
+columns:
+  - title: Example
+    key: example
+  - title: Description
+    key: desc
+rows:
+  - example: "[Auto mode](./examples/auto/)"
+    desc: "Run the Solace Consume plugin in `auto` mode, letting the plugin automatically determine the mode from the client request."
+  - example: "[Polling mode](./examples/polling/)"
+    desc: "Run the Solace Consume plugin in its default polling mode."
+  - example: "[Server-sent events mode](./examples/server-sent-events)"
+    desc: "Stream messages using [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events)."
+  - example: "[WebSocket mode](./examples/websocket/)"
+    desc: "Stream messages over a [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API) connection."
+  - example: "[SSL connections](./examples/ssl/)"
+    desc: "Enable SSL validation."
+  - example: "[URI capture](./examples/uri-capture/)"
+    desc: "Capture the URI of the request and use it as the message destination name."
+{% endtable %}
+
+## Implementation details
+
+The plugin supports the following modes of operation:
+* `polling`: Check for messages at set intervals instead of waiting for a push notification (default)
+* `auto`: Determine the mode automatically from the client request
+* `server-sent-events`: Stream messages using [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events)
+* `websocket`: Stream messages over a [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API) connection
```

Comment:
```suggestion
* `polling`: Checks for messages at set intervals instead of waiting for a push notification (default)
* `auto`: Determines the mode automatically from the client request
* `server-sent-events`: Streams messages using [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events)
* `websocket`: Streams messages over a [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API) connection
```